packages/@react-stately/layout/src/ListLayout.ts

```diff
@@ -297,7 +364,8 @@ export class ListLayout<T> extends Layout<Node<T>> implements KeyboardDelegate,
     layoutInfo.allowOverflow = true;
     layoutInfo.estimatedSize = isEstimated;
     return {
-      layoutInfo
+      layoutInfo,
+      validRect: layoutInfo.rect
```

Comment:
Just wanna verify I understand why `validRect` here for the computed item is `layoutInfo.rect` versus being `layoutInfo.rect.intersection(this.validRect)` like it is for the section. If I'm following the code flow here, these `validRect` calculated for the sections/items are used to determine if the cached layout node can be reused (aka `cached.validRect.containsRect(cached.layoutInfo.rect.intersection(this.validRect))` at line 237). Setting this `validRect` as `layoutInfo.rect` would mean we can use the cached information for the layout node so long as it is intersecting the list's `validRect`. However, the computed section has `validRect: layoutInfo.rect.intersection(this.validRect)` since we would need to recompute the section's children's layout that may now be in the `validRect` if a different part of the section is now within the list's `validRect`? 

Math/visualization is hard :(

Reply:
Yeah, an item is never partially valid, but a section can be (because it has children that might be outside the valid rect).